terraform-tests/mssql_test.go

```diff
@@ -63,4 +64,200 @@ var _ = Describe("mssql", Label("mssql-terraform"), Ordered, func() {
 			}))
 		})
 	})
+
+	Context("properties that pass-through without validation", func() {
+		When("mssql_version is passed", func() {
+			It("it passes-through without validation", func() {
+				// It would be possible to validate the engine version before apply using
```

Comment:
we could, we used to have enum in the yaml for engine versions. We removed them consciously to avoid having to release and ask to upgrade each time a new version comes out in one of the several services we have in all of the three brokers.

Reply:
We also try to delegate validation to the provider / AWS API whenever possible, we don't want to rewrite all of their logic.

Reply:
Having safety-nets written as postconditions can ensure no partial infrastructure is created during an unsuccessful apply.